File: 백준/1번문제/최원준/Q9012.java
// https://www.acmicpc.net/problem/
package 백준;

/*


1. 아이디어 :

2. 시간복잡도 :
O( )
3. 자료구조 :


 */
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

class Q9012 {
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    public static String check(String s) {
        Stack<Character> stack = new Stack<>();
        for (char c : s.toCharArray()) {
            if (stack.isEmpty() && c == ')') return "NO";

            if (stack.isEmpty() || c == '(') {
                stack.push(c);
            } else {
                if (stack.peek() == '(') {
                    stack.pop();
                } else {
                    stack.push(c);
                }
            }
        }
        return stack.size()==0? "YES":"NO";
    }
    public static void main(String[] args) throws IOException {

        int n = Integer.parseInt(br.readLine());
        for (int i=0; i<n; i++) {
            System.out.println(check(br.readLine()));
        }

    }
}


/*

import sys
#sys.setrecursionlimit(1000000)
# input = sys.stdin.readline
input = lambda: sys.stdin.readline().rstrip()

def solution(s):
    stack = []
    for c in s:
        if not stack and c == ")": return "NO";

        if not stack or c == "(":
            stack.append(c)
        else: # )
            if stack[-1] == "(":
                stack.pop()
    return "YES" if len(stack)==0 else "NO"



if __name__ == '__main__':
    n = int(input())
    for i in range(n):
        s = input().strip()
        print(solution(s))

 */

File: 백준/1번문제/S0okJu/Q9012.py
"""

1. 아이디어 : 스택에는 ( 만 들어가야 하며 완성된 VPS에는 남아있는 (가 없어야 한다.

2. 시간복잡도 :
O(n^2)
3. 자료구조 : Stack

"""

def solution(ps):
    stack = []
    for j in range(len(ps)):
        if ps[j] == '(':
            stack.append('(')
        elif ps[j] == ')':
            if len(stack) != 0:
                stack.pop()
            else:
                return False
                
    
    if len(stack) > 0:
        return False
    else:
        return True
                
                
if __name__ == '__main__':
    t = int(input())
    for i in range(t):
        ps = input()
        if (solution(ps)==False):
            print("NO")
        else:
            print("YES")
    